xarray/coding/strings.py

```diff
@@ -103,16 +103,21 @@ def encode(self, variable, name=None):
         dims, data, attrs, encoding = unpack_for_encoding(variable)
         if data.dtype.kind == 'S' and encoding.get('dtype') is not str:
             data = bytes_to_char(data)
-            dims = dims + ('string%s' % data.shape[-1],)
+            if 'char_dim_name' in encoding.keys():
+                dims = dims + (encoding['char_dim_name'],)
+            else:
+                default_char_dim_name = 'string%s' % data.shape[-1]
+                dims = dims + (default_char_dim_name,)
+                encoding['char_dim_name'] = default_char_dim_name
```

Comment:
I think this line is causing a set of failures. I think this line is desirable. I could be wrong.

Reply:
line 111, to be clear.

Reply:
The convention we have (which is apparently enforced by tests) is that when you use an encoding you remove it from the `encoding` dict. That way we catch cases where you have a typo, e.g., `char_dimension_name` instead of `char_dim_name`.

So this section should instead probably be something like:
```python
char_dim_name = encoding.pop(
    'char_dim_name', 'string%s' % data.shape[-1])
dims = dims + (char_dim_name,)
```

Reply:
Great reason, this solves all the errors I was seeing locally. THanks